storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -90,32 +92,40 @@ func (s *peersSource) Read(
 		shardRetrieverMgr block.DatabaseShardBlockRetrieverManager
 		persistFlush      persist.Flush
 		incremental       = false
+		seriesCachePolicy = s.opts.ResultOptions().SeriesCachePolicy()
 	)
-	if opts.Incremental() {
+	if opts.Incremental() && seriesCachePolicy != series.CacheAll {
 		retrieverMgr := s.opts.DatabaseBlockRetrieverManager()
 		persistManager := s.opts.PersistManager()
-		if retrieverMgr != nil && persistManager != nil {
-			s.log.WithFields(
-				xlog.NewField("namespace", namespace.String()),
-			).Infof("peers bootstrapper resolving block retriever")
 
-			r, err := retrieverMgr.Retriever(nsMetadata)
-			if err != nil {
-				return nil, err
-			}
+		// Neither of these should ever happen
+		if seriesCachePolicy != series.CacheAll && retrieverMgr == nil {
+			panic("Tried to perform incremental flush without retriever manager")
```

Comment:
Let's use `logger.Fatalf(...)` instead, also nit: use lowercase to begin message.

Reply:
done